frontend/gateway/pb/gateway.proto

```diff
@@ -90,6 +90,8 @@ message ResolveImageConfigRequest {
 message ResolveImageConfigResponse {
 	string Digest = 1 [(gogoproto.customtype) = "github.com/opencontainers/go-digest.Digest", (gogoproto.nullable) = false];
 	bytes Config = 2;
+	bytes Manifest = 3;
```

Comment:
Having these 2 levels of indirection in here doesn't really cover all the cases. There could be nested indexes, and in attestations PR you are adding descriptors for the attestation manifests etc that can't be accessed(and therefore copied by imagetools) with this method.

I wonder if instead we should just always return the root manifest(the one that digest is pointing to). Client can parse it and make more specific digest results next to resolve the internal objects. This would make `Config` value optional so maybe there should be a flag in request side that controls if empty config should be an error or not.